2-WebApp-graph-user/2-5-HybridFlow/README.md

```diff
@@ -0,0 +1,771 @@
+  
+  ```json
+  {
+    // ... 
+    "AzureAd": {
+      // ... 
+      "Certificate":  {
+        "SourceType":  "Path",
+        "CertificateDiskPath":  "<PATH_TO_YOUR_CERTIFICATE_FILE>",
+        "CertificatePassword":  "<CERTIFICATE_PASSWORD>"
+      }
+    } 
+  }
+  ```
+  
+  #### Get certificate from Key Vault
+  
+  It's also possible to get certificates from an [Azure Key Vault](https://docs.microsoft.com/azure/key-vault/general/overview). Replace the values in the `Certificate` key of the `appsettings.json` file with the snippet shown below also replacing `<YOUR_KEY_VAULT_URL>` with the URL of the Key Vault holding your certificate and `<YOUR_KEY_VAULT_CERTIFICATE_NAME>` with the name of that certificate as shown in your Key Vault. If you created the application with the `Configure.ps1` script found in the `AppCreationScripts-withCert` a **pfx** file called **HybridFlowCert.pfx** will be generated that is associated with the certificate that can be used as a credential for your app. If you like, you can load that certificate into a Key Vault and then access that Key Vault to use as a credential for your application.
+
+  ```json
+  {
+    // ... 
+    "AzureAd": {
+      // ... 
+      "Certificate":  {
+        "SourceType":  "KeyVault",
+        "KeyVaultUrl":  "<YOUR_KEY_VAULT_URL>",
+        "KeyVaultCertificateName":  "<YOUR_KEY_VAULT_CERTIFICATE_NAME>"
+      }
+    }
+  }
+  ```
+  
+  3. If you had set `ClientSecret` previously, change its value to empty string, `""`.
+
+</details>
+
+### Step 4: Running the sample
+
+ For command line run the next commands:
+
+```console
+    cd 2-WebApp-graph-user\2-5-HybridFlow
+    dotnet run
+```
+
+## Using the sample
+
+<details>
+ <summary>Expand to see how to use the sample</summary>
+ Open your web browser and navigate to `https://localhost:7089`. You should see the application home page with a link for **Home**, **Privacy** and **Sign in**. Click the **Sign in** link and you'll be redirected to the Microsoft login page. Sign in using an user account in your tenant. After you sign in the client side `MSAL.js` client will receive an **auth** code from the server that will be exchanged for an authorization token and cached immediately in the browser.
+
+1. Click on `Profile` tab to see the signed in user's information and contacts. This information is retrieved using the `MSAL.js` library in the browser.
+
+Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.
+
+[Consider taking a moment to share your experience with us.](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRz0h_jLR5HNJlvkZAewyoWxUNEFCQ0FSMFlPQTJURkJZMTRZWVJRNkdRMC4u)
+</details>
+
+## About the code
+
+<details>
+ <summary>Expand to see more information about the code</summary>
+
+## Server setup
```

Comment:
>Server setup [](http://example.com/codeflow?start=3&length=12)

Hmm.. Should this be like "Code in Asp.net core web app". Server-side gives a very different meaning